internal/service/privatelinkendpointservice/resource_privatelink_endpoint_service_test.go

```diff
@@ -17,6 +18,37 @@ func TestAccNetworkRSPrivateLinkEndpointServiceAWS_Complete(t *testing.T) {
 	resource.Test(t, *testCase)
 }
 
+func TestAccNetworkRSPrivateLinkEndpointServiceAWS_Failed(t *testing.T) {
+	acc.SkipTestForCI(t) // needs AWS configuration
+	var (
+		resourceSuffix = "test"
+		resourceName   = fmt.Sprintf("mongodbatlas_privatelink_endpoint_service.%s", resourceSuffix)
+
+		awsAccessKey = os.Getenv("AWS_ACCESS_KEY_ID")
+		awsSecretKey = os.Getenv("AWS_SECRET_ACCESS_KEY")
+
+		providerName = "AWS"
+		projectID    = os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+		region       = os.Getenv("AWS_REGION")
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheck(t); acc.PreCheckAwsEnvPrivateLinkEndpointService(t) },
+		CheckDestroy:             checkDestroy,
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		ExternalProviders:        acc.ExternalProvidersOnlyAWS(),
+		Steps: []resource.TestStep{
+			{
+				Config: configFailAWS(
+					awsAccessKey, awsSecretKey, projectID, providerName, region, resourceSuffix,
+				),
+				Check:       resource.TestCheckResourceAttr(resourceName, "error_message", "privatelink endpoint is in a failed state: Interface endpoint vpce-11111111111111111 was not found."),
```

Comment:
if plan fails, is it useful to have this attribute error_message that won't be able to be used?

Reply:
sorry, maybe I'm not understanding this test. Is the Check useful in the test when we also have a ExpectError? We are expecting an error so, so for ExpectError to be pass the plan or apply must fail, so in that case I think it doesn't matter what the attribute values are as it failed.

Reply:
That's true, the check is more to outline that the error_message field has been correctly populated but ultimately isn't needed in this test since we expect an error. This check can be removed.